src/function/table/table_scan.cpp

```diff
@@ -404,14 +404,20 @@ void ExtractConjunctionAnd(ConjunctionAndFilter &filter, BoundColumnRefExpressio
 	vector<reference<InFilter>> in_filters;
 
 	for (idx_t i = 0; i < filter.child_filters.size(); i++) {
-		if (filter.child_filters[i]->filter_type == TableFilterType::CONSTANT_COMPARISON) {
-			auto &comparison = filter.child_filters[i]->Cast<ConstantFilter>();
+		auto &child_filter = *filter.child_filters[i];
+		switch (child_filter.filter_type) {
+		case TableFilterType::CONSTANT_COMPARISON: {
+			auto &comparison = child_filter.Cast<ConstantFilter>();
 			comparisons.push_back(comparison);
-			continue;
+			break;
 		}
-
-		if (filter.child_filters[i]->filter_type == TableFilterType::OPTIONAL_FILTER) {
-			auto &optional_filter = filter.child_filters[i]->Cast<OptionalFilter>();
+		case TableFilterType::CONJUNCTION_AND: {
+			auto &conjunction = child_filter.Cast<ConjunctionAndFilter>();
+			ExtractConjunctionAnd(conjunction, bound_ref, expressions);
```

Comment:
If we recurse here and use the index scan on the nested conjunction, are we still evaluating the original (top level) expression on the result of the index scan?

Reply:
I think so! Each level of `CONJUNCTION_AND` is added to the `vector<unique_ptr<Expression>> &expressions`, so we AND them all together in the end

Reply:
Ah, perfect! :)